src/main/webapp/script.js

```diff
@@ -88,3 +90,43 @@ function getJsonIfOk(response) {
   }
 }
 
+/**
+ * Handles response by checking it and converting it to text.
+ */
+function handleTextResponse(response) {
+  if (!response.ok) {
+    throw new Error("Response error while fetching data: " +
+        response.status + " (" + response.statusText + ")");
+  }
+  return response.text();
+}
+
+/**
+ * Loads the user's email and sets the source
+ * of the calendar iframe.
+ */
+function loadCalendar() {
+  fetch("/user")
+    .then(handleTextResponse)
+    .then(setCalendar)
+    .catch(handleNetworkError);
+}
+
+/**
+ * Sets the source of the calendar iframe including
+ * the email of the user.
+ */
+function setCalendar(email) {
+  const calendarIframe = document.getElementById("calendar");
+  calendarIframe.src = "https://calendar.google.com/calendar/embed?src=" +
```

Comment:
You could use the more readable template strings here instead. (Supported everywhere but IE)

https://developer.mozilla.org/en-US/docs/Web/JavaScript/Reference/Template_literals

Reply:
That's a cool feature.